lib/child-writer/index.js

```diff
@@ -113,25 +113,44 @@ exports.write = (image, drive, options) => {
     emitter.emit('error', error);
   };
 
-  ipc.server.on('error', emitError);
-  ipc.server.on('message', (data) => {
-    let message = null;
-
+  /**
+   * @summary Bridge robot message to the child writer caller
+   * @function
+   * @private
+   *
+   * @param {String} message - robot message
+   *
+   * @example
+   * bridgeRobotMessage(robot.buildMessage('foo', {
+   *   bar: 'baz'
+   * }));
+   */
+  const bridgeRobotMessage = (message) => {
     try {
-      message = robot.parseMessage(data);
-    } catch (error) {
-      return emitError(error);
-    }
+      const parsedMessage = robot.parseMessage(message);
+
+      // The error object is decomposed by the CLI for serialisation
+      // purposes. We compose it back to an `Error` here in order
+      // to provide better encapsulation.
+      if (robot.getCommand(parsedMessage) === 'error') {
+        emitError(robot.recomposeErrorMessage(parsedMessage));
+      } else {
+
+        // These are lighweight accessor methods for
+        // the properties of the parsed message
+        const messageCommand = robot.getCommand(parsedMessage);
+        const messageData = robot.getData(parsedMessage);
```

Comment:
These could be moved a few lines higher up, to eliminate the duplicated `robot.getCommand(parsedMessage)` ?